runtime/src/utils/locators.js

```diff
@@ -7,15 +7,20 @@ function normalizeSlash(s) {
 }
 
 // Default locator definitions.
-export function defaultURLLocator(base) {
+export function defaultURLLocator(bases = []) {
   // normalize trailing slash
-  base = normalizeSlash(base);
+  bases = bases.map(normalizeSlash);
   return async (raw) => {
-    if (base === null) {
-      return null;
-    }
     let module = raw.replace(/^GRAIN\$MODULE\$/, '');
-    return readURL(base + "/" + module + ".wasm");
+    for (const base of bases) {
+      let fullpath = base + "/" + module + ".wasm";
+      try {
+        return await readURL(fullpath);
```

Comment:
Not an async function

Reply:
Awkward.

Reply:
Oh yes it is: https://github.com/grain-lang/grain/blob/master/runtime/src/core/grain-module.js#L189

Reply:
Sorry, I meant the function that you are using the `await` inside, which shouldn't work if it isn't `async`, right?

Reply:
There's an `async` on line 13. A little hard to see.

Reply:
Ah, didn't realize this function returned another function.